flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/SetOpRewriteUtil.scala

```diff
@@ -81,7 +81,7 @@ object SetOpRewriteUtil {
     val fieldTypes = logicalType.getChildren.map(fromLogicalTypeToTypeInfo).toArray
     val tf = new ReplicateRows(fieldTypes)
     val resultType = fromLegacyInfoToDataType(new RowTypeInfo(fieldTypes, fieldNames))
-    val function = new TypedFlinkTableFunction(tf, resultType)
+    val function = new TypedFlinkTableFunction(tf, fieldNames, resultType)
```

Comment:
This should be first commit.

Reply:
Good catch. However `SetOpRewriteUtil` is a new class, which I found compile error after rebase code, so The update not in first commit...